hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/strategy/ClusteringPlanStrategy.java

```diff
@@ -56,6 +59,37 @@
   private final transient HoodieEngineContext engineContext;
   private final HoodieWriteConfig writeConfig;
 
+  /**
+   * Check if the given class is deprecated.
+   * If it is, then try to convert it to suitable one and update the write config accordingly.
+   * @param config write config
+   * @return class name of clustering plan strategy
+   */
+  public static String checkAndGetClusteringPlanStrategy(HoodieWriteConfig config) {
+    String className = config.getClusteringPlanStrategyClass();
+    String sparkSizeBasedClassName = "org.apache.hudi.client.clustering.plan.strategy.SparkSizeBasedClusteringPlanStrategy";
```

Comment:
Sure, will reuse constants. 

And about these legacy class name, I'll keep them as it is because they are not in `HoodieClusteringConfig`. Moving them to the config file may confuse users.